File: solutions/python/0643-maximum-average-subarray-i.py
class Solution:
    def findMaxAverage(self, nums: List[int], k: int) -> float:
        max_avg = float('-inf') 
        window_sum = 0 
        for i in range(len(nums)): 
            window_sum += nums[i] 
            if i >= k - 1: 
                max_avg = max(max_avg, window_sum / k) 
                window_sum -= nums[i - k + 1]  
        return max_avg 
                                                                                                

File: solutions/cpp/0643-maximum-average-subarray-i.cpp
class Solution {
public:
    double findMaxAverage(vector<int>& nums, int k) {
        double window_sum = 0;
        for (int i = 0; i < k; i++) {
            window_sum += nums[i];
        }
        double max_avg = window_sum / k;
        
        for (int i = k; i < nums.size(); i++) {
            window_sum += nums[i] - nums[i - k];
            max_avg = max(max_avg, window_sum / k);
        }
        return max_avg;
    }
};
